tactics/tactics.ml

```diff
@@ -1455,7 +1455,8 @@ exception IsNonrec
 let is_nonrec mind = (Global.lookup_mind (fst mind)).mind_finite == Declarations.BiFinite
 
 let find_ind_eliminator ind s gl =
-  let gr = lookup_eliminator ind s in
+  let env = Proofview.Goal.env gl in
```

Comment:
Just a comment, be careful here as it is not always the case the this env will be the global env as before; in particular see #6523 and the actual code that creates `Goal.env`:
```
  let gmake_with info env sigma goal state =
    { env = Environ.reset_with_named_context (Evd.evar_filtered_hyps info) env ;
```
`tclEnv` should be better I think, but I am not sure.

Reply:
Doesn't matter, `lookup_eliminator` only looks at the global table. Moreover, calls to `tclENV` should be an exception, most of the time the relevant data is the goal context.

Reply:
Indeed it would be way more civilized to have a type of global environements and a type of proof environments (that extends the latter with the proof context)

Reply:
@maximedenes indeed you could pass something of type `Environ.globals` if it wasn't for the resolver; but something weird is going on lookup eliminator, why does the nametab need to get involved with a name that came out of the kernel?